tests/IntegrationTests/Compression.fs

```diff
@@ -0,0 +1,80 @@
+module Pulsar.Client.IntegrationTests.Compression
+
+open System
+open Expecto
+open Pulsar.Client.Api
+open FSharp.Control.Tasks.V2.ContextInsensitive
+open System.Threading.Tasks
+open Serilog
+open Pulsar.Client.IntegrationTests.Common
+open Pulsar.Client.Common
+
+[<Tests>]
+let tests =
+
+    let codecs =
+        [|
+            CompressionType.None
+            CompressionType.LZ4
+            CompressionType.Snappy
+            CompressionType.ZLib
+            CompressionType.ZStd
+        |]
+
+    let sendReceive enableBatching (compressionType : CompressionType) = async {
+
+        let singleOrBatched = if enableBatching then "batched" else "single"
+
+        Log.Debug("Started Send and receive {0} compressed message using '{1}'", singleOrBatched, compressionType)
+
+        let client = getClient()
+        let topicName = "public/default/topic-" + Guid.NewGuid().ToString("N")
+
+        let! producer =
+            ProducerBuilder(client)
+                .Topic(topicName)
+                .ProducerName("compression-single")
```

Comment:
What does single mean? I suggest renaming to compression-producer

Reply:
Done.